Add start script; Allow configuration through env vars

dkarnutsch opened the pull request:
As we use s2i, we need NPM to run the start script or we would need to overwrite the entrypoint. In the future this could be replaced by directly calling node (without NPM)

dkarnutsch committed:
Add start script

As we use s2i, we need NPM to run the start script or we would need to overwrite the entrypoint.
In the future this could be replaced by directly calling node (without NPM)

nsams commented on package.json:
```diff
@@ -10,7 +10,8 @@
         "test": "jest",
-        "test:watch": "jest --watch"
+        "test:watch": "jest --watch",
+        "start": "node dist/index.js $ORIGIN_URL --port $PORT --cacheSizeLimitHint $CACHE_SIZE_LIMIT_HINT --cacheDir $CACHE_DIR"
```
hm, what if an env var is not set (or a new one is added?)

alternatively you could pack them all into one xxx_OPTS env (except ORIGIN_URL)

dkarnutsch committed:
Allow configuration through environment variables

dkarnutsch commented:
@nsams you are right. I've changed it to support environment variables and removed the arguments from the start script.

I also needed to switch for _origin_ from argument to option to support environment variables.

Now it can be either started using environment variables or command line arguments.

nsams commented on src/index.ts:
```diff
@@ -1,20 +1,20 @@
-    .option("--cacheSizeLimitHint <megabytes>", "maximum cache size hint", "500")
-    .option("--cacheDir <dir>", "directory cache files will be written into", "cache")
-    .action((origin: string, { port, cacheDir, cacheSizeLimitHint }: { port: string; cacheDir: string; cacheSizeLimitHint: string }) => {
+    .addOption(new Option("--origin <origin>", "Origin server URL").env("ORIGIN_URL").makeOptionMandatory())
```
why change that to an option? Imho having this important, required part as argument is a nice cli interface

dkarnutsch replied:
As stated in https://github.com/vivid-planet/swr-cache-proxy/pull/18#issuecomment-1744734689

> I also needed to switch for origin from argument to option to support environment variables.

Alternatively, we could revert this change and use
```
"start": "node dist/index.js $ORIGIN_URL
```
